src/core/server/http/router/route.ts

```diff
@@ -24,11 +24,74 @@ import { ObjectType } from '@kbn/config-schema';
+ * @public
+ */
+export type RouteContentType =
+  | 'application/json'
+  | 'application/*+json'
+  | 'application/octet-stream'
+  | 'application/x-www-form-urlencoded'
+  | 'multipart/form-data'
+  | 'text/*';
+
+/**
+ * Additional body options for a route
+ * @public
+ */
+export interface RouteConfigOptionsBody {
+  /**
+   * A string or an array of strings with the allowed mime types for the endpoint. Use this settings to limit the set of allowed mime types. Note that allowing additional mime types not listed
+   * above will not enable them to be parsed, and if parse is true, the request will result in an error response.
+   *
+   * Default value: allows parsing of the following mime types:
+   * * application/json
+   * * application/*+json
+   * * application/octet-stream
+   * * application/x-www-form-urlencoded
+   * * multipart/form-data
+   * * text/*
+   */
+  accepts?: RouteContentType | RouteContentType[] | string | string[];
+
+  /**
+   * Limits the size of incoming payloads to the specified byte count. Allowing very large payloads may cause the server to run out of memory.
+   *
+   * Default value: The one set in the kibana.yml config file under the parameter `server.maxPayloadBytes`.
+   */
+  maxBytes?: number;
+
+  /**
+   * The processed payload format. The value must be one of:
+   * * 'data' - the incoming payload is read fully into memory. If parse is true, the payload is parsed (JSON, form-decoded, multipart) based on the 'Content-Type' header. If parse is false, a raw
+   * Buffer is returned.
+   * * 'stream' - the incoming payload is made available via a Stream.Readable interface. If the payload is 'multipart/form-data' and parse is true, field values are presented as text while files
+   * are provided as streams. File streams from a 'multipart/form-data' upload will also have a hapi property containing the filename and headers properties. Note that payload streams for multipart
+   * payloads are a synthetic interface created on top of the entire multipart content loaded into memory. To avoid loading large multipart payloads into memory, set parse to false and handle the
+   * multipart payload in the handler using a streaming parser (e.g. pez).
+   *
+   * Default value: 'data', unless no validation.body is provided in the route definition. In that case the default is 'stream' to alleviate memory pressure.
+   */
+  output?: 'data' | 'stream';
+
+  /**
+   * Determines if the incoming payload is processed or presented raw. Available values:
+   * * true - if the request 'Content-Type' matches the allowed mime types set by allow (for the whole payload as well as parts), the payload is converted into an object when possible. If the
+   * format is unknown, a Bad Request (400) error response is sent. Any known content encoding is decoded.
+   * * false - the raw payload is returned unmodified.
+   * * 'gunzip' - the raw payload is returned unmodified after any known content encoding is decoded.
```

Comment:
I see only one usage. Can we switch to `true` instead? `'gunzip'` looks to generic and hapi specific

Reply:
Unfortunately, I don't think we can use `true` in [that one usage](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/elasticsearch/lib/create_proxy.js#L31). It's related to the (multi-search endpoint)[https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html] in ElasticSearch where we are receiving NDJSON format in the form of
```
header\n
body\n
header\n
body\n
```
Example:
```
{"index" : "test"}
{"query" : {"match_all" : {}}, "from" : 0, "size" : 10}
{"index" : "test", "search_type" : "dfs_query_then_fetch"}
{"query" : {"match_all" : {}}}
{}
{"query" : {"match_all" : {}}}

{"query" : {"match_all" : {}}}
{"search_type" : "dfs_query_then_fetch"}
{"query" : {"match_all" : {}}}
```
I don't think HAPI knows how to parse that, and using  `false`, we may risk passing through to ElasticSearch a compressed payload that it may not understand.

Reply:
I see. Optional: Shouldn't we rename it to something like `try`. Not to expose specific implementation?